### PR TITLE
RR-797: Update conversation endpoint implementation

### PR DESCRIPTION
**Pull request by leejacobson-moj**
## Overview

Adds the implementation for the update conversation endpoint.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaConversationPersistenceAdapter.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ class JpaConversationPersistenceAdapter(
     val conversationEntity = conversationRepository.findByReference(updateConversationDto.reference)
     return if (conversationEntity != null) {
       conversationEntityMapper.updateEntityFromDto(conversationEntity, updateConversationDto)
+      conversationEntity.updateLastUpdatedAt() // force the main Induction's JPA managed fields to update
```
Comment (leejacobson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaConversationPersistenceAdapter.kt:30]: After our conversation @nathanrussell-moj-digital I took a look at Inductions to see if it had a similar problem and found this line was forcing the JPA managed fields to update even when the parent entity itself had no updates. This seemed like the easiest solution to the problem I was having so replicated here.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt [hunk 2]
```diff
@@ -29,12 +32,26 @@ class ConversationController(
   @ResponseStatus(HttpStatus.CREATED)
   @PreAuthorize(HAS_EDIT_AUTHORITY)
   @Transactional
-  fun createInduction(
+  fun createConversation(
```
Comment (leejacobson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt:35]: Missed this copy and paste typo in the last PR.

File: src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml [hunk 3]
```diff
@@ -2731,8 +2731,6 @@ components:
           type: string
           description: The identifier of the prison that the prisoner is currently resident at.
           example: BXI
-        type:
-          $ref: '#/components/schemas/ConversationType'
```
Comment (leejacobson-moj) [inline src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:2735]: I think I included this by accident. There's no user need to update the conversation type (at least not currently).

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**